scripts/main.js

```diff
@@ -308,9 +314,89 @@ window.addEventListener('WebChannelMessageToContent', function (event) {
       selectNextItem(-1);
     }
   }
+
+  // collecting session metrics on open and closed message types
+  if (message.type === 'popupopen') {
```

Comment:
These messages aren't reliable (they are fired at the wrong time, and duplicates are fired), I don't think the data gathered by listening for these events would be useful.

I'm planning to remove the empty popup code from the prototype, which should make these more reliable. We can just add this for now and expect to get less-than-helpful metrics until I fix the gecko bugs, let me know what you'd like to do.

If the underlying goal is to identify when the user interacts with the popup, we could add mousemove listeners and also hook into existing key and click event listeners.